pkg/netpol/connlist/exposure_analysis_test.go

```diff
@@ -0,0 +1,232 @@
+	exposedToEntireCluster: true,
+	potentialConn:          common.MakeConnectionSet(true),
+}
+
+var peerExposedToEntireClusterOnTCP8050 *xgressExposure = &xgressExposure{
+	exposedToEntireCluster: true,
+	potentialConn:          newTCPConnWithPort(8050),
+}
+
+func newTCPConnWithPort(port int) *common.ConnectionSet {
+	conn := common.MakeConnectionSet(false)
+	portSet := common.MakePortSet(false)
+	portSet.AddPort(intstr.FromInt(port))
+	conn.AddConnection(v1.ProtocolTCP, portSet)
+	return conn
+}
+
+func newExpDataWithLabelAndTCPConn(key, val string, portNum int) *xgressExposure {
+	conn := common.MakeConnectionSet(true)
+	if portNum != -1 {
+		conn = newTCPConnWithPort(portNum)
+	}
+	return &xgressExposure{
+		exposedToEntireCluster: false,
+		namespaceLabels:        map[string]string{key: val},
+		podLabels:              map[string]string{},
+		potentialConn:          conn,
+	}
+}
+
+// TestExposureBehavior tests the behavior of exposure analysis
+func TestExposureBehavior(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		testName                       string
+		expectedNumRepresentativePeers int
+		expectedLenOfExposedPeerList   int
+		wl1ExpDataInfo                 expectedPeerResultInfo
+		wl2ExpDataInfo                 expectedPeerResultInfo
+	}{
+		{
+			testName:                       "test_allow_all", // only workload-a in manifests
+			expectedNumRepresentativePeers: 0,
+			expectedLenOfExposedPeerList:   1,
+			// workload 1 unsecure exposed to all other end-points in the world
+			wl1ExpDataInfo: expectedPeerResultInfo{
+				isIngressProtected:     true,
+				isEgressProtected:      true,
+				lenIngressExposedConns: 1, // entire cluster conns
+				ingressExp: []*xgressExposure{
+					peerExposedToEntireCluster,
+				},
+				lenEgressExposedConns: 1,
+				egressExp: []*xgressExposure{
+					peerExposedToEntireCluster,
+				},
+			},
+		},
+		{
+			testName:                       "test_allow_all_in_cluster", // only workload-a in manifests
+			expectedNumRepresentativePeers: 0,
+			expectedLenOfExposedPeerList:   1,
+			// workload 1 unsecure exposed to all other end-points in the cluster
+			wl1ExpDataInfo: expectedPeerResultInfo{
+				isIngressProtected:     true,
+				isEgressProtected:      true,
+				lenIngressExposedConns: 1,
+				lenEgressExposedConns:  1,
+				ingressExp: []*xgressExposure{
+					peerExposedToEntireClusterOnTCP8050,
+				},
+				egressExp: []*xgressExposure{
+					peerExposedToEntireCluster,
+				},
+			},
+		},
+		{
+			testName:                       "test_matched_and_unmatched_rules",
+			expectedNumRepresentativePeers: 1,
+			expectedLenOfExposedPeerList:   2,
+			// workload 1 is protected only on ingress direction and exposed unsecure to entire cluster
+			wl1ExpDataInfo: expectedPeerResultInfo{
+				isIngressProtected:     true,
+				isEgressProtected:      false,
+				lenIngressExposedConns: 1, // connection to unknown dst is contained in entire cluster's conn
+				ingressExp: []*xgressExposure{
```

Comment:
this test includes a "connection" containment 
the pod is exposed to any ns with `foo.com/managed-state` on `TCP8050`
and to `entire-cluster` on `TCP8050` : so only the entire cluster conn is in the final result

Reply:
ok, good.
do you also have example where both are involved in the final result? (for example the namespace connection is TCP 8050-8051)?

Reply:
no, added one